persistence/blockstore/block_store.go

```diff
@@ -0,0 +1,65 @@
+package blockstore
+
+//go:generate mockgen -source=$GOFILE -destination=../types/mocks/blockstore/block_store_mock.go github.com/pokt-network/pocket/persistence/types BlockStore
+
+import (
+	"fmt"
+
+	"github.com/pokt-network/pocket/persistence/kvstore"
+)
+
+type BlockStore interface {
+	kvstore.KVStore
+}
+
+var _ BlockStore = &blockStore{}
+
+// BlockStore wraps a KVStore to manage savepoint and rollback behavior.
```

Comment:
This is great but complimentary to my request above. The way I think about it is: if we run `godoc`, this is internal and won't show up, but the above interface will.